src/App.js

```diff
@@ -53,24 +57,32 @@ class Landing extends React.Component {
   }
 }
 
-function SignIn() {
+class SignIn extends React.Component {
   // uses setLocalToken to set token to local storage when button is clicked
-  return (
-    <form className="shadow bg-white pa2" action="./add-item">
-      <h1 className="b f1">Welcome to your smart shopping list!</h1>
-      <p className="f3">Tap “Create shopping list” to get started.</p>
-      <button onClick={setLocalToken} className="bg-green ph2 pv1 white f2 b">
-        Create shopping list
-      </button>
-      <p className="f5 gray">
-        You can also{' '}
-        <a className="black" href="/">
-          join an existing shopping list
-        </a>
-        .
-      </p>
-    </form>
-  );
+  handleClickSetLocalToken = () => {
+    setLocalToken();
+  };
+  render() {
+    return (
+      <form className="shadow bg-white pa2" action="./add-item">
+        <h1 className="b f1">Welcome to your smart shopping list!</h1>
+        <p className="f3">Tap “Create shopping list” to get started.</p>
+        <button
+          onClick={this.handleClickSetLocalToken}
+          className="bg-green ph2 pv1 white f2 b"
+        >
+          Create shopping list
+        </button>
+        <p className="f5 gray">
+          You can also{' '}
+          <a className="black" href="/join-existing">
+            join an existing shopping list
+          </a>
+          .
+        </p>
+      </form>
```

Comment:
In the future (let's keep this as is because it's working fine!), let's try a couple of things:

1. Remove the `onClick` event from the `button` element, and instead add it to the `form` element as a `onSubmit` event. As we talked in the first TCL call, thanks to @msholty-fd!, adding the event to the `onSubmit` will give your users a better experience because every time they hit `enter` the form will be submitted. Right now, the user will have to either click or tab + space the button to get this working.
Again, we don't need to change this as I believe it's out of scope and was already introduced in a previous PRs

2. I'm not really sure about the usage of `action="./add-item"` in the form tag, what this is doing is that it's redirecting right after sending the form (without any validation) and it always asumes the happy path. You have two options:
a. You can use the [useHistory hook](https://reacttraining.com/react-router/web/api/Hooks/usehistory), but that would mean that you have to replace the your class component with a functional component
b. You can use [withRouter](https://reacttraining.com/react-router/web/api/withRouter) to get `history` via `this.props.history`
Once you're done with that, you can use the `history.push` method to navigate to the next page, `history.push("/home");`

I'd like us to change this before merging **if you're 👍 with it and have time to implement it**, if not, I'd suggest moving forward as this is working right now!

Reply:
I really agree with point 2 here @alejandronanez -

I believe having an `onSubmit` handler is a good idea, but the `action` attribute makes this form behave as if we weren't working within the confines of a JS application. The `action` attribute will `POST` the form's data to that url, but since we are in a JS application we can validate it immediately here.

Reply:
Thanks so much! Spent the better part of this morning familiarizing myself with Hooks 💪🏽

I changed the `SignIn` class component to a functional one, imported `useHistory` from `react-router-dom` and added an `onSubmit` event handler to perform both setting a token in local storage and routing to `./add-item`.

Additionally, I wrote up a short summary of what I learned:
> Before React Hooks the biggest difference between a class and functional component was syntax, state and lifecycle hooks. If you wanted to create a dynamic component, one that used state and lifecycle to change states, you’d use a class component. Functional components could only use state if you lifted state up to the parent component and passed it down to the functional component via props.
> 
> The choice was clear: Class components to take advantage of state and lifecycle methods. Otherwise, use functional components to take advantage of their clear and readable syntax and performance benefits.
> 
> This left us with three points of frustration: 
> 1. It was hard to reuse stateful logic between components
> 2. Complex components became hard to understand
> 3. Classes confused both people and machines
> 
> React 16.8 introduced Hooks which enabled functional components to use state and lifecycle events. This meant you could advantage of the clear syntax and performance of functional components and the state and lifecycle methods that were once only available for class components.